src/kevedit/screen.c

```diff
@@ -230,12 +228,20 @@ char* filenamedialog(char* initname, char* extension, char* prompt, int askoverw
 	pathof(path, initname, path_size);
 
 	/* if extension is given, remove extension from buffer */
-	if (extlen > 0)
+	if (extlen > 0) {
 		for (i = strlen(filenamebuffer) - 1; i >= 0; i--)
 			if (filenamebuffer[i] == '.') {
 				filenamebuffer[i] = '\x0';
 				break;
 			}
+		/* truncate extension for printing if necessary */
+		if (extlen > (sizeof(extprinted) - 1)) {
+			strncpy(extprinted, extension, sizeof(extprinted) - 1 - 3);
```

Comment:
This will blow up unless `extprinted` already happens to be zero-filled, since `strncpy` won't add the null terminator if it's after `n` characters

Reply:
Is this better?

Reply:
Perfect! Thanks!